modules/mod_ginger_activity/services/service_ginger_activity_inbox.erl

```diff
@@ -0,0 +1,84 @@
+%% @doc Retrieve activity notifications.
+-module(service_ginger_activity_inbox).
+
+-svc_needauth(true).
+
+-export([
+    process_post/2,
+    process_get/2
+]).
+
+-include_lib("zotonic.hrl").
+-include_lib("../include/ginger_activity.hrl").
+
+%% @doc Get activity stream inbox for a user.
+process_get(_ReqData, Context) ->
+    Activities = m_ginger_activity_inbox:for_user(z_acl:user(Context), Context),
+    Json = #{
+        <<"@context">> => <<"https://www.w3.org/ns/activitystreams">>,
+        <<"type">> => <<"Collection">>,
+        <<"items">> => [to_json(A, Context) || A <- Activities]
+    },
+    jsx:encode(Json).
+
+%% @doc Delete an item from the user's activity stream inbox.
+process_post(#wm_reqdata{method = 'DELETE'}, Context) ->
+    User = z_acl:user(Context),
+    case z_context:get_q(id, Context) of
+        undefined ->
+            %% Delete complete user's activity stream.
+            m_ginger_activity_inbox:delete(User, Context);
+        Id ->
+            IntId = z_convert:to_integer(Id),
+            m_ginger_activity_inbox:delete(User, IntId, Context)
+    end,
+    <<>>.
```

Comment:
I was wondering if we should return the data of the deleted item here, so ran a quick check on what is the common/best practice - there doesn't seem to be a hard one; https://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api#http-status for instance suggests returning http status 204 (success, no content) on delete without a body.